pkg/sql/schema_resolver.go

```diff
@@ -300,12 +300,15 @@ func (sr *schemaResolver) GetTypeDescriptor(
 	ctx context.Context, id descpb.ID,
 ) (tree.TypeName, catalog.TypeDescriptor, error) {
 	tc := sr.descCollection
-	desc, err := tc.GetImmutableTypeByID(ctx, sr.txn, id, tree.ObjectLookupFlags{})
+	flags := sr.CommonLookupFlagsRequired()
+	flags.ParentID = sr.typeResolutionDbID
```

Comment:
Thanks for doing this, I'd have forgotten about doing this for types if it's me.